src/main/java/com/nutomic/syncthingandroid/util/HTTPS.java

```diff
@@ -0,0 +1,136 @@
+ *
+ */
+public class HTTPS {
+
+    /**
+     * Taken from: http://janis.peisenieks.lv/en/76/english-making-an-ssl-connection-via-android/
+     *
+     */
+    private static class CustomSSLSocketFactory extends SSLSocketFactory {
+        SSLContext sslContext = SSLContext.getInstance("TLS");
+
+        public CustomSSLSocketFactory(String httpsCertPath, KeyStore truststore)
+                throws NoSuchAlgorithmException, KeyManagementException,
+                KeyStoreException, UnrecoverableKeyException {
+            super(truststore);
+            TrustManager tm = new CustomX509TrustManager(httpsCertPath);
+            sslContext.init(null, new TrustManager[] { tm }, null);
+        }
+
+        public CustomSSLSocketFactory(SSLContext context)
+                throws KeyManagementException, NoSuchAlgorithmException,
+                KeyStoreException, UnrecoverableKeyException {
+            super(null);
+            sslContext = context;
+        }
+
+        @Override
+        public Socket createSocket(Socket socket, String host, int port,
+                                   boolean autoClose) throws IOException, UnknownHostException {
+            return sslContext.getSocketFactory().createSocket(socket, host, port,
+                    autoClose);
+        }
+
+        @Override
+        public Socket createSocket() throws IOException {
+            return sslContext.getSocketFactory().createSocket();
+        }
+    }
+
+    private static class CustomX509TrustManager implements X509TrustManager {
+
+        private String mHttpsCertPath;
+
+        public CustomX509TrustManager(String httpsCertPath) {
+            mHttpsCertPath = httpsCertPath;
+        }
+
+        @Override
+        public void checkClientTrusted(X509Certificate[] chain, String authType)
+                throws CertificateException {
+        }
+
+        @Override
+        public void checkServerTrusted(java.security.cert.X509Certificate[] certs,
+                                       String authType) throws CertificateException {
+            // Here you can verify the servers certificate. (e.g. against one which is stored on mobile device)
+            InputStream inStream = null;
+            try {
+                inStream = new FileInputStream(mHttpsCertPath);
+                CertificateFactory cf = CertificateFactory.getInstance("X.509");
+                X509Certificate ca = (X509Certificate)
+                        cf.generateCertificate(inStream);
+                for (X509Certificate cert : certs) {
+                    cert.verify(ca.getPublicKey());
+                }
+            } catch (Exception e) {
+                throw new IllegalArgumentException("Untrusted Certificate!");
+            } finally {
+                try {
+                    if (inStream != null)
+                        inStream.close();
+                } catch (IOException e) {
```

Comment:
The difference is when you have eg a NullPointerException in that code and won't notice it.